### PR TITLE
Add bidirectional pagination support for VirtualizedList

### PR DESCRIPTION
**Pull request by roryabraham**
This PR ports over changes from our fork of React Native, specifically these PRs:

- https://github.com/Expensify/react-native/pull/17/files
- https://github.com/Expensify/react-native/pull/18/files

### CODE REVIEW

File: packages/react-native-web/src/vendor/react-native/VirtualizedList/index.js [hunk 1]
```diff
@@ -1566,26 +1638,72 @@ class VirtualizedList extends React.PureComponent<Props, State> {
...
-      // Only call onEndReached once for a given content length
       this._sentEndForContentLength = this._scrollMetrics.contentLength;
       onEndReached({distanceFromEnd});
-    } else if (distanceFromEnd > threshold) {
-      // If the user scrolls away from the end and back again cause
-      // an onEndReached to be triggered again
-      this._sentEndForContentLength = 0;
+    }
+
+      // Next check if the user just scrolled within the start threshold
```
Comment (AndrewGable) [inline packages/react-native-web/src/vendor/react-native/VirtualizedList/index.js]: NAB - Spacing here is a bit inconsistent

**Comment by AndrewGable** [comment]
Few questions:
1. How did we test this?
2. Can we build these changes and publish to our own fork? Might be out of scope, but just curious.

**Comment by roryabraham** [comment]
> Can we build these changes and publish to our own fork? Might be out of scope, but just curious.

Yes! Was planning on doing this in a follow-up

File: packages/react-native-web/src/vendor/react-native/VirtualizedList/index.js (hunk 1 above)
Comment (roryabraham) [reply packages/react-native-web/src/vendor/react-native/VirtualizedList/index.js]: Fixed.

**Comment by roryabraham** [comment]
> How did we test this

No testing was done locally; all these changes were ported directly from our react-native fork. Since the file kept in react-native-web is an almost-exact copy of the react-native equivalent, I figured it would be safe enough to port our changes over

If you have ideas for how to test this in isolation, I'm open to that. Maybe we could put something in the react-native-web-examples folder?

**Review by AndrewGable — APPROVED** [review]

**Comment by AndrewGable** [comment]
Merging since other reviewer is OOO.

**Pull request merged**